src/transaction/log_recovery_redo_parallel.hpp

```diff
@@ -216,25 +219,38 @@ namespace cublog
 	   * NOTE: '*_locked_*' functions are supposed to be called from within locked
 	   * areas with respect to the resources they make use of
 	   */
-	  ux_redo_job_base do_locked_find_job_to_consume_and_mark_in_progress (
+	  ux_redo_job_deque do_locked_find_job_to_consume_and_mark_in_progress (
 		  const std::lock_guard<std::mutex> &a_consume_lockg,
 		  const std::lock_guard<std::mutex> &a_in_progress_lockg);
 
 	  /* NOTE: '*_locked_*' functions are supposed to be called from within locked
 	   * areas with respect to the resources they make use of
 	   */
-	  void do_locked_mark_job_in_progress (
+	  void do_locked_mark_job_deque_in_progress (
 		  const std::lock_guard<std::mutex> &a_in_progress_lockg,
-		  const ux_redo_job_base &a_job);
+		  const ux_redo_job_deque &a_job_deque);
 
 	private:
-	  /* two queues are internally managed and take turns at being either
+	  /* two maps of queues are internally managed and take turns at being either
 	   * on the producing or on the consumption side
 	   */
-	  ux_redo_job_deque *m_produce_queue;
-	  mutable std::mutex m_produce_queue_mutex;
-	  ux_redo_job_deque *m_consume_queue;
-	  std::mutex m_consume_queue_mutex;
+	  vpid_ux_redo_job_deque_map_t *m_produce;
```

Comment:
```suggestion
	  vpid_ux_redo_job_deque_map_t *m_produce_jobs;
```

Reply:
:) I simplified it because in one refactoring it is a `deque` in another one it will be a `map`, aso.

Reply:
Done